lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -121,12 +121,13 @@ IRGenDebugInfo::IRGenDebugInfo(const IRGenOptions &Opts,
   }
 
   unsigned Lang = llvm::dwarf::DW_LANG_Swift;
-  std::string Producer = version::getSwiftFullVersion();
+  std::string Producer = version::getSwiftFullVersion(
+    Opts.EffectiveLanguageVersion);
   bool IsOptimized = Opts.Optimize;
   StringRef Flags = Opts.DWARFDebugFlags;
-  unsigned Major, Minor;
-  std::tie(Major, Minor) = version::getSwiftNumericVersion();
-  unsigned MajorRuntimeVersion = Major;
+
+  assert(!Opts.EffectiveLanguageVersion.empty());
```

Comment:
@adrian-prantl Oh, weird, I wonder why I thought this had to do with the associated ObjC runtime. Ok, that makes sense; reverting that change then. Thanks!